conn.go

```diff
@@ -584,6 +617,60 @@ type callReq struct {
 	timer *time.Timer
 }
 
+type writeCoalescer struct {
+	w       io.Writer
+	timeout time.Duration
+
+	cond    *sync.Cond
+	mu      sync.Mutex
+	buffers net.Buffers
+
+	// result of the write
+	err error
+}
+
+func (w *writeCoalescer) flush() {
+	if w.timeout > 0 {
+		type deadliner interface {
+			SetWriteDeadline(time.Time) error
+		}
+		w.w.(deadliner).SetWriteDeadline(time.Now().Add(w.timeout))
+	}
+
+	defer w.mu.Unlock()
+	w.mu.Lock()
+
+	if len(w.buffers) == 0 {
+		return
+	}
+
+	// TODO: do we care about the number of bytes written? Given
+	// we are going to do a fanout n is useless and according to
+	// the docs WriteTo should return 0 and err or bytes written and
+	// no error.
```

Comment:
Considering the `WriteTo` seems to be pretty clear in it's writing the whole buffer until drained or erroring out, I think we really don't care about the number of bytes written. We can and do deduct that from the buffer's length. So maybe just leave the comment why we don't store that without the TODO mark.